src/Psalm/Type/Atomic/TValueOfArray.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Psalm\Type\Atomic;
+
+use Psalm\Type\Atomic;
+
+/**
+ * Represents a value of an array.
+ */
+class TValueOfArray extends Atomic
+{
+    /** @var TClassConstant|TKeyedArray|TList|TArray */
+    public $type;
+
+    /**
+     * @param TClassConstant|TKeyedArray|TList|TArray $type
+     */
+    public function __construct(Atomic $type)
+    {
+        $this->type = $type;
+    }
+
+    public function getKey(bool $include_extra = true): string
+    {
+        return 'value-of<' . $this->type . '>';
+    }
+
+    public function __toString(): string
+    {
+        return $this->getKey();
+    }
+
+    /**
+     * @param  array<lowercase-string, string> $aliased_classes
+     */
+    public function toPhpString(
+        ?string $namespace,
+        array $aliased_classes,
+        ?string $this_class,
+        int $analysis_php_version_id
+    ): ?string {
+        return null;
+    }
+
+    public function canBeFullyExpressedInPhp(int $analysis_php_version_id): bool
+    {
+        return false;
+    }
+
+    public function getAssertionString(bool $exact = false): string
+    {
+        return 'mixed';
```

Comment:
I'm not sure about this, I just adopted from old `TValueOfClassConstant`. The default implementation calls `getId()` what should be a return value of this method?

Reply:
it should be fine. getAssertionString is used when we want to refine a type from another. returning mixed is like saying there's nothing to refine, I'm not sure we can do a lot more.

Possibly you could infer the actual type of the key or values and if the type is simple enough (like `string`) returning it here, but I doubt this will be very used